include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -71,22 +66,10 @@ class pairwise_alignment_algorithm_banded :
      *
      * Initialises the algorithm given the user settings from the alignment configuration object.
      *
-     * \throws seqan3::invalid_alignment_configuration if the given band settings are invalid.
+     * \throws seqan3::invalid_alignment_configuration.
      */
     pairwise_alignment_algorithm_banded(alignment_configuration_t const & config) : base_algorithm_t(config)
-    {
-        using seqan3::get;
-
-        auto band = get<seqan3::align_cfg::band_fixed_size>(config);
-
-        lower_diagonal = band.lower_diagonal.get();
-        upper_diagonal = band.upper_diagonal.get();
-
-        if (upper_diagonal < lower_diagonal || upper_diagonal < 0 || lower_diagonal > 0)
-            throw invalid_alignment_configuration{"The selected band [" + std::to_string(lower_diagonal) + ":" +
-                                                  std::to_string(upper_diagonal) + "] is not supported because it "
-                                                  "does not completely encloses both sequences."};
-    }
```

Comment:
Okay you removed that here again. Without much thought, is the sequence invariant check more general than this?\\

I think the `upper_diagonal < lower_diagonal` is the only invariant that is a user error, and we should warn him about this before starting the computation.

Reply:
No, the user error would also to provide a band which does not cover the first cell and a global alignment is requested. A local alignment or a semi-global alignment would be valid though.

Reply:
> I think the upper_diagonal < lower_diagonal is the only invariant that is a user error, and we should warn him about this before starting the computation.

But we can't do it inside of configuration object because the parameters can be set separately. So it throws during configuration.